### PR TITLE
Add Scala Steward worfklow

### PR DESCRIPTION
**Pull request by MancunianSam**
This runs Scala Steward once per day on each of our repos.
If there are updates, it creates a PR using the Jenkins user for the new version.

### CODE REVIEW

File: .github/workflows/scala_steward.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 0 * * 0'
```
Comment (TomJKing) [inline .github/workflows/scala_steward.yml:4]: Does this match up with the dependabot schedule, ie once a day? They should be running the same to keep things consistent.

File: repos.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# repos.md
```
Comment (TomJKing) [inline repos.md:1]: The documentation will need to be updated to remind that this list will need updating if a new scala repo is added: https://github.com/nationalarchives/tdr-dev-documentation/blob/master/manual/create-repo.md

**Review by TomJKing — COMMENTED** [review]
Looks good. Couple of minor comments.

File: .github/workflows/scala_steward.yml (hunk 1 above)
Comment (MancunianSam) [reply .github/workflows/scala_steward.yml:4]: Yeah, this is once a day. It follows this https://pubs.opengroup.org/onlinepubs/9699919799/utilities/crontab.html#tag_20_25_07

File: repos.md (hunk 2 above)
Comment (MancunianSam) [reply repos.md:1]: I'll put a PR in for that in a sec.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**